File: Prepare/Algorithms/Warmup/10-Time-Conversion/Python3/main.py
import re

time = input()
time_match = re.compile(r"^([01]?[0-9]):([0-5][0-9]):([0-5][0-9])(PM|AM)$")
hours = int(time_match.search(time).group(1))
minutes = int(time_match.search(time).group(2))
seconds = int(time_match.search(time).group(3))
ampm = time_match.search(time).group(4)

if ampm == "PM":
    if hours != 12:
        hours = int(hours) + 12
elif hours == 12:
        hours = 0
    
print(f"{hours:02}:{minutes:02}:{seconds:02}")

File: Prepare/Algorithms/Warmup/10-Time-Conversion/C/main.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int hours, minutes, seconds;
    char am_pm;
    scanf("%d:%d:%d%c", &hours, &minutes, &seconds, &am_pm);

    if (am_pm == 'P') {
        if (hours != 12) {
            hours += 12;
        }
    } else {
        if (hours == 12) {
            hours = 0;
        }
    }

    printf("%02d:%02d:%02d", hours, minutes, seconds);

    return 0;
}